src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -464,6 +464,11 @@ def config_source_control(resource_group_name, name, repo_url, repository_type=N
             sc = SourceControl(location, name='GitHub', token=git_token)
             client.update_source_control('GitHub', sc)
 
+        # service doesn't support update for being a less likely scenario, but no error is emitted.
+        # we work around it by deleting the existing binding.
+        if show_source_control(resource_group_name, name, slot).repo_url:
+            delete_source_control(resource_group_name, name, slot)
```

Comment:
I guess my question is, since we are deleting what was there as part of the workaround, are we preserving the old values? If I ran this once using the non-default parameters and then ran it again just to change the URL to another branch, I would expect those other parameter values to stay the same way they were (with the non-defaults).  Would this work for that scenario, or would the object be recreated with the defaults?  If so, I would view that as a bug.